packages/api/src/@types/PuzzleDecorator.ts

```diff
@@ -0,0 +1,8 @@
+type PuzzleDecoratorItem = {
+  decorator: 'CLAWBACK';
+  clawbackTimelock: number;
+};
+
+type PuzzleDecorator = PuzzleDecoratorItem[];
```

Comment:
Best practice in Typesscript is define single item and not whole array as type.
You should always use type with array in your function.